File: acceptance/b33-b3k/13-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string r) {
        int answer = 0;
        map<char, int> map;

        map['I'] = 1;
         map['L'] = 50;
        map['C'] = 100;
       
        map['V'] = 5;
        map['X'] = 10;
         map['D'] = 500;
        map['M'] = 1000;
       
        
       
        
        for(int i=0; i<r.length(); i++){
            if(map[r[i]] < map[r[i+1]]){
                answer -= map[r[i]];
            }
            else{
                answer += map[r[i]];
            }
        }
        return answer;

    }
};

File: acceptance/Prajwal-kp/13-roman-to-integer.py
class Solution(object):
    def romanToInt(self, s):
        roman_dict = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
        }
        result = 0
        prev_val = 0
        for i in range(len(s) - 1, -1, -1):
            val = roman_dict[s[i]]
            if val < prev_val:
                result -= val
            else:
                result += val
            prev_val = val
        return result

solution = Solution()
output = solution.romanToInt("VI")
print(output)    # Output: 6